src/main/java/codechicken/lib/asm/RedirectorTransformer.java

```diff
@@ -0,0 +1,190 @@
+
+
+    static {
+        Collections.addAll(redirectedFields,
+                     "pipeline"
+                    ,"model"
+                    ,"firstVertexIndex"
+                    ,"lastVertexIndex"
+                    ,"vertexIndex"
+                    ,"baseColour"
+                    ,"alphaOverride"
+                    ,"useNormals"
+                    ,"computeLighting"
+                    ,"useColour"
+                    ,"lightMatrix"
+                    ,"vert"
+                    ,"hasNormal"
+                    ,"normal"
+                    ,"hasColour"
+                    ,"colour"
+                    ,"hasBrightness"
+                    ,"brightness"
+                    ,"side"
+                    ,"lc"
+
+                );
+        Collections.addAll(redirectedSimpleMethods,
+                "reset",
+                "pullLightmap",
+                "pushLightmap",
+                "setDynamic",
+                "draw");
+        Collections.addAll(
+                redirectedMethods,
+                "setPipeline",
+                "bindModel",
+                "setModel",
+                "setVertexRange",
+                "render",
+                "runPipeline",
+                "writeVert",
+                "setNormal",
+                "setColour",
+                "setBrightness",
+                "startDrawing");
+
+        cstPoolParser = new ClassConstantPoolParser(RenderStateClass);
+    }
+
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass) {
+        if (!cstPoolParser.find(basicClass)) {
+            return basicClass;
+        }
+
+
+        final ClassReader cr = new ClassReader(basicClass);
+        final ClassNode cn = new ClassNode();
+        cr.accept(cn, 0);
+        boolean changed = false;
+
+        for (MethodNode mn : cn.methods) {
+            for (AbstractInsnNode node : mn.instructions.toArray()) {
+                if (node.getOpcode() == Opcodes.GETSTATIC && node instanceof FieldInsnNode fNode) {
+                    if(redirectedFields.contains(fNode.name) && fNode.owner.equals(RenderStateClass)) {
+                        mn.instructions.insertBefore(fNode, new MethodInsnNode(Opcodes.INVOKESTATIC, fNode.owner, "instance", "()Lcodechicken/lib/render/CCRenderState;"));
+                        fNode.setOpcode(Opcodes.GETFIELD);
+                        changed = true;
+                    }
+                }
+                else if (node.getOpcode() == Opcodes.PUTSTATIC && node instanceof FieldInsnNode fNode) {
+                    if(redirectedFields.contains(fNode.name) && fNode.owner.equals(RenderStateClass)) {
+                        final String fName = "set" + StringUtils.capitalize(fNode.name) + "Static";
+                        final String fDesc = "(" + fNode.desc + ")V";
+                        mn.instructions.set(node, new MethodInsnNode(Opcodes.INVOKESTATIC, fNode.owner, fName, fDesc));
```

Comment:
I'd prefer to insert a INVOKESTATIC instance() then SWAP before PUTSTATIC. This works too, but adds quite a bit of boilerplate method.